climpred/relative_entropy.py

```diff
@@ -0,0 +1,301 @@
+    if curv:  # if curvilinear lon(x,y), lat(x,y) data inputs
+        wgts = None
+    else:
+        coslat = np.cos(np.deg2rad(anom_x.coords['lat'].values))
+        wgts = np.sqrt(coslat)[..., np.newaxis]
+
+    if isinstance(control, xr.Dataset):  # EOF requires xr.dataArray
+        control = control.to_array().squeeze()
+
+    if 'member' in control.dims:  # LENS
+        # stack member and initialization into time dim, make time first
+        non_spatial_control_dims = list(
+            set(control.dims).intersection(['time', 'member']))
+
+        transpose_dims = list(control.dims)
+        transpose_dims.remove('member')
+        dims = tuple(transpose_dims)
+
+        base_to_calc_eofs = control.stack(
+            new=tuple(non_spatial_control_dims)).rename({'new':
+                                                         'time'}).set_index({'time': 'time'}).transpose(*dims)
+    else:
+        # PM_control
+        base_to_calc_eofs = control.rename({'initialization': 'time'})
+
+    solver = Eof(base_to_calc_eofs, weights=wgts)
+
+    re_leadtime_list = []
+    lead_times = initialized.time.values[:ntime]
+    initializations = initialized.initialization.values
+    # DoTo: parallelize this double loop
+    for init in initializations:  # loop over initializations
+        rl, sl, dl = ([] for _ in range(3))  # lists to store results in
+        for t in lead_times:  # loop over lead time
+            # P_b base distribution
+            pc_b = solver.projectField(anom_b.sel(initialization=init, time=t)
+                                             .drop('time')
+                                             .rename({'member': 'time'}),
+                                       neofs=neofs, eofscaling=0,
+                                       weighted=False)
+
+            mu_b = pc_b.mean('time')
+            sigma_b = xr.DataArray(np.cov(pc_b.T))
+
+            # P_x initialization distribution
+            pc_x = solver.projectField(anom_x.sel(initialization=init, time=t)
+                                             .drop('time')
+                                             .rename({'member': 'time'}),
+                                       neofs=neofs, eofscaling=0,
+                                       weighted=False)
+
+            mu_x = pc_x.mean('time')
+            sigma_x = xr.DataArray(np.cov(pc_x.T))
+
+            r, d, s = _relative_entropy_formula(sigma_b, sigma_x, mu_x, mu_b,
+                                                neofs)
+
+            rl.append(r)
+            sl.append(s)
+            dl.append(d)
+
+        re_leadtime_list.append(xr.Dataset({'R': ('time', rl),
+                                            'S': ('time', sl),
+                                            'D': ('time', dl)}))
+
+    re = xr.concat(re_leadtime_list, dim='initialization').assign(
+        initialization=initializations, time=lead_times)
+
+    return re
+
+
+def bootstrap_relative_entropy(initialized, control, sig=95,
```

Comment:
(1) This currently doesn't work for the LENS-style setup. Since we already have the uninitialized ensemble, can we just calculate relative_entropy from there?  Is there a simple way to do this?

(2) Would it make sense to call this from `compute_relative_entropy` every time, or have a keyword defaulted to `True` on calling this? I imagine that nearly every user of this would want to know significance bounds for their relative entropy calculation.

Reply:
I think we can decide those things once relative_entropy is added to the PredictionEnsemble class.

Reply:
Agreed. Reopened the issue for this.